pkg/operation/botanist/secrets.go

```diff
@@ -96,7 +96,17 @@ func (b *Botanist) GenerateAndSaveSecrets(ctx context.Context) error {
 					}
 				}
 			} else {
-				if err := b.cleanupTunnelSecrets(ctx, &gardenerResourceDataList, vpnseedserver.DeploymentName, vpnseedserver.VpnShootSecretName, vpnseedserver.VpnSeedServerTLSAuth); err != nil {
+				if err := b.cleanupSecrets(ctx, &gardenerResourceDataList, vpnseedserver.DeploymentName, vpnseedserver.VpnShootSecretName, vpnseedserver.VpnSeedServerTLSAuth); err != nil {
+					return err
+				}
+			}
+
+			if settings := b.Seed.GetInfo().Spec.Settings; settings == nil ||
+				settings.DependencyWatchdog == nil ||
+				settings.DependencyWatchdog.Probe == nil ||
+				!settings.DependencyWatchdog.Probe.Enabled {
```

Comment:
I would suggest to extract this in a method similar to what was done for the OwnerChecks: https://github.com/gardener/gardener/blob/6095397d3bc75852bf4ad5b86ea5c9dfc9080a20/pkg/apis/core/v1beta1/helper/helper.go#L885-L887
Then we could reuse it in wanted_secrets.go file

I guess the same could be done for the DWD endpoint settings